packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.jsx

```diff
@@ -42,7 +42,19 @@ export const StorefrontPreview = ({enabled = true, getToken}) => {
     ) : null
 }
 
+StorefrontPreview.defaultProps = {
+    enabled: true
+}
+
 StorefrontPreview.propTypes = {
     enabled: PropTypes.bool,
-    getToken: PropTypes.func
+    // a custom prop type function to only require this prop if enabled is true.
+    getToken: function (props, propName, componentName) {
+        if (
+            props['enabled'] === true &&
+            (props[propName] === undefined || typeof props[propName] !== 'function')
+        ) {
+            return new Error(`${propName} is required for ${componentName} when enabled is true`)
```

Comment:
Minor: not only the prop is required, it should also be a function.
```suggestion
            return new Error(`${propName} is a required function for ${componentName} when enabled is true`)
```